integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/database/DatabaseOperationChecker.java

```diff
@@ -0,0 +1,135 @@
+import static com.sequenceiq.redbeams.api.model.common.Status.DELETE_COMPLETED;
+import static com.sequenceiq.redbeams.api.model.common.Status.DELETE_IN_PROGRESS;
+import static com.sequenceiq.redbeams.api.model.common.Status.DELETE_REQUESTED;
+import static com.sequenceiq.redbeams.api.model.common.Status.PRE_DELETE_IN_PROGRESS;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.ProcessingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.util.wait.service.ExceptionChecker;
+import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.responses.DatabaseServerV4Response;
+import com.sequenceiq.redbeams.api.model.common.Status;
+
+public class DatabaseOperationChecker<T extends DatabaseWaitObject> extends ExceptionChecker<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseOperationChecker.class);
+
+    @Override
+    public boolean checkStatus(T waitObject) {
+        String crn = waitObject.getCrn();
+        Status desiredStatus = waitObject.getDesiredStatus();
+        try {
+            DatabaseServerV4Response database = waitObject.getEndpoint().getByCrn(crn);
+            if (database == null) {
+                throw new TestFailException(String.format("'%s' database was not found.", crn));
+            }
+            String name = database.getName();
+            Status status = database.getStatus();
+            LOGGER.info("Waiting for the '{}' state of '{}' '{}' database. Actual state is: '{}'", desiredStatus, name, crn, status);
+            if (isDeletionInProgress(status) || status == DELETE_COMPLETED) {
+                LOGGER.error("Database '{}' '{}' has been getting terminated (status:'{}'), waiting is cancelled.", name, crn, status);
+                throw new TestFailException(String.format("Database '%s' '%s' has been getting terminated (status:'%s'), waiting is cancelled.", name, crn,
+                        status));
+            }
+            if (waitObject.isFailed(status)) {
+                LOGGER.error("Database '{}' '{}' is in failed state (status:'{}'), waiting is cancelled.", name, crn, status);
+                throw new TestFailException(String.format("Database '%s' '%s' is in failed state. Status: '%s' statusReason: '%s'",
+                        name, crn, status, database.getStatusReason()));
+            }
+            if (desiredStatus.equals(status)) {
+                return true;
+            }
+        } catch (Exception e) {
+            StringBuilder builder = new StringBuilder("Database has been failed. Also failed to get database status: ")
+                    .append(System.lineSeparator())
+                    .append(e.getMessage())
+                    .append(System.lineSeparator())
+                    .append(e);
+            LOGGER.error(builder.toString());
```

Comment:
Why call it twice? I guess the compiler will optimize it anyway but still strange a bit. I saw it earlier as well in other files. Can you please introduce a `message` variable?